src/c++/simulator/boundaries.hpp

```diff
@@ -0,0 +1,44 @@
+﻿
+#pragma once
+
+#include <simulator/geometry.hpp>
+#include <simulator/particle.hpp>
+#include <simulator/my_types.hpp>
+
+#include <iostream>
+
+struct Boundary : public IParticle
+{
+public:
+	virtual Point get_point();
+	virtual Vector get_unitary_normal();
+	Entity get_type() override;
+	virtual void print();
+	double get_radius();
+
+public:
+	Geometry* geometry; 
```

Comment:
I think we should use `unique_ptr` if this `Geometry` instance will be unique.

Reply:
Even better if we can make it `Geometry` (no pointers at all)

Reply:
Much better.